api/src/controllers/team.js

```diff
@@ -313,6 +336,68 @@ async function updateTeam(req, res, next) {
   }
 }
 
+/**
+ * Clear the team Data from the database on /team/:team_id
+ * @param {} req request object, containing team id in the url
+ * @param {*} res response object, the deleted team document
+ * @returns 200: teamn is deleted
+ * @returns 404: team is not found
+ * @returns 400: team id is not in a valid hexadecimal format
+ */
+async function resetTeamData(req, res, next) {
+  try {
+    const { teamId } = req.params;
+    const { isDeleteFlag } = req.body;
+    await Achievement.deleteMany({ teamId });
```

Comment:
What about deleting the other models? Homepage, theme, etc …

Reply:
Tbh I think you should make this functionality as a database trigger rather than encapsulate all this logic in the controller. I can help with that if you need

Reply:
> Sorry but I’m a little confused. According to the PR desc. I’d imagine there’d be at most two endpoints: one for deleting an entire teams data (clean wipe) and one to delete their data but retain their deployed website. Could you explain the reasoning behind adding more loosely tied functionality in this PR, like resetting webpages, etc?

So it was initally exactly this , but I refactored the functionality based on the reviews i got throughout PI3